src/arrays/qucoeffs.jl

```diff
@@ -0,0 +1,69 @@
+import Base: transpose,
+    ctranspose,
+    conj,
+    size,
+    ndims,
+    length,
+    getindex,
+    setindex!
+
+############
+# QuCoeffs #                 
+############
+    abstract ConjBool{Conj}
+    abstract TranBool{Tran}
```

Comment:
The bool wrapper types are there for type stability purposes. Otherwise, the only type stable way to construct an instance of QuCoeffs would be to manually declare them as parameters (which is obviously annoying since you'd have to manually declare the other variables as well) or have the constructor always set those parameters to false (which is limiting/less flexible).

As for only indicating duality - that's how I started, but then I ran into some of the problems discussed in  JuliaLang/julia#6837. Mainly, [figuring out which operations we want to result in copies and which we want to result in views](https://github.com/JuliaLang/julia/pull/6837#issuecomment-43904028). If we just had a duality indicator instead of separate conjugate and transposition indicators, then the behavior gets wonky/unintuitive - a transpose would cause a copy, a conjugation would cause a copy, but the ctranspose would cause a view. Thus, doing `conj(transpose(arr))` would be a double copy operation, while `ctranspose(arr)` would be a no-copy operation, which seems ugly/confusing to me.

I'm of the opinion that it makes the most sense for all three of the above operations to "agree" to return a view, or a copy. I obviously went with the view here, and tried to reduce the inefficiencies associated computing conjugation on the fly by having it be a no-op for non-`Complex` types.